vector/src/main/java/im/vector/app/features/home/room/detail/timeline/item/PollItem.kt

```diff
@@ -44,60 +44,44 @@ abstract class PollItem : AbsMessageItem<PollItem.Holder>() {
     @EpoxyAttribute
     var pollSent: Boolean = false
 
+    @EpoxyAttribute
+    var totalVotesText: String? = null
+
+    @EpoxyAttribute
+    var optionViewStates: List<PollOptionViewState>? = null
+
     override fun bind(holder: Holder) {
         super.bind(holder)
         val relatedEventId = eventId ?: return
 
         renderSendState(holder.view, holder.questionTextView)
 
         holder.questionTextView.text = pollContent?.pollCreationInfo?.question?.question
+        holder.totalVotesTextView.text = totalVotesText
 
-        holder.optionsContainer.removeAllViews()
+        val cachedViews = mutableMapOf<String, PollOptionItem>()
+        holder.optionsContainer.children.filterIsInstance<PollOptionItem>().forEach { existingPollItemView ->
+            cachedViews[existingPollItemView.getTag(STUB_ID)?.toString() ?: ""] = existingPollItemView
+        }
 
-        val isEnded = pollResponseSummary?.isClosed.orFalse()
-        val didUserVoted = pollResponseSummary?.myVote?.isNotEmpty().orFalse()
-        val totalVotes = pollResponseSummary?.totalVotes ?: 0
-        val winnerVoteCount = pollResponseSummary?.winnerVoteCount
+        holder.optionsContainer.removeAllViews()
```

Comment:
Maybe avoid to remove (and re-add later) all the View? The number of views may differ.

Reply:
I just tested it and it is not working to remove items one by one before adding. Ends up with duplicated views.

Reply:
Say you have 3 options to display. If there is:
- 0 existing views: create 4 views and add it to the parent (this will be the most cases)
- 4 existing views: just remove one, and do not add any
- 3 existing views: do nothing
- 2 existing views: create one new and add it to the parent

Reply:
I see your point now. It will be an interesting experiment for me.